### DISCUSSION TITLE
Charging behavior for multiple batteries

### DISCUSSION THREAD

**Discussion by coca7** [Louisvdw/dbus-serialbattery#117] [General]
I am wondering how the driver will handle multiple batteries in parallel when it is released.  I have three in parallel and one of my batteries is getting out of sync with the other two.  It is about 20% off now. 
![D289CBDB-20B0-4576-B7A3-4A30D9EFECAD_1_105_c](https://user-images.githubusercontent.com/2902641/164069024-597227b7-2310-40bf-a64a-0b13b6ebe878.jpeg)

**Comment by Louisvdw** [comment]
The driver cannot handle anything electrical, also it only reading the BMS values and not changing them.
So the multi battery feature will only accumulate the values to show one large battery to the venus/victron system. It will do that with what is available. 

For instance it will create a weighted average of the SOC and voltage value and accumulate the current. |
In a case like this it will show that some charging capacity is still available until all BMS show 100% SOC, but some alerts or errors will influence that as well.
When one BMS blocks charge this will change the charge current setting and similar discharge and will also show modules blocking charge. 
Electricly that should balance out as the lowest voltage will pull up/down the other voltages, and so the current will follow.
Naturally this all works best when all the banks are the same and balanced, but it should handle it they are not as well

**Comment by ghost** [comment]
Just out of curiosity, are your batteries connected with equal length wires
to a main distribution bus bar, so that all the loads are distributed
equally among the batteries?

Kind regards,
Shaun Schmidt

**Reply by coca7** [reply]
Yes, all the wires are the same length.  The batteries were also at the same state of charge when I started.  However, it is my understanding that even with the same length wires its is hard for batteries in parallel to maintain the same state of charge.  Some manufacturers like Renege have some software in the batteries to actively keep the batteries the same.  It's my understanding that without that there will always be one batteries in the setup that gets hit harder than the others.

**Comment by Louisvdw** [comment]
Some banks would see more current draw than others, but I would expect that to be still close to each other if they started the same. IT has to do with minor differences in the resistance and capacity of the chemical cells.
You did check that the settings in each BMS are the same like the battery capacity value.

**Reply by bergyla** [reply]
Should the current release support multiple BMS ( different DALY ) ? I get only one to see.
However the USB-UART isolating adapters (FTDI232R) all reporting different Serials and getting reliable COM-Ports on Windows...

First post states when its released but with screenshot of multiple BMS...

**Reply by Louisvdw** [reply]
> Should the current release support multiple BMS

Not yet https://github.com/Louisvdw/dbus-serialbattery/issues/8
